Source/deviceinfo/DeviceInfo.cpp

```diff
@@ -181,10 +184,23 @@ class DeviceInfoLink : public WPEFramework::RPC::SmartInterfaceType<WPEFramework
             _identifierInterface = nullptr;
         }
         BaseClass::Close(WPEFramework::Core::infinite);
+        _singleton = nullptr;
     }
     static DeviceInfoLink& Instance()
     {
-        return WPEFramework::Core::SingletonType<DeviceInfoLink>::Instance();
+        static DeviceInfoLink *instance = new DeviceInfoLink;
```

Comment:
This is indeed what I would expect :-) A static on the "instance = new DeviceInfoLink" so that it is only called the first time it is called (Lazy Creation)